File: chapter1/q1/q1.py
def isUnique(text : str) -> bool:
    charSet = set()
    for c in text:
        if c in charSet:
            return False
        charSet.add(c)
    return True

def isUniqueSorted(text : str) -> bool:
    text = sorted(text)
    for i in range(1, len(text)):
        if text[i] == text[i-1]:
            return False
    return True

text = "a"
print(isUnique(text))
print(isUniqueSorted(text)) 

File: chapter1/q1/q1.cpp
#include <iostream>
#include <set>
#include <string>
#include <algorithm>
using namespace std;

bool isUnique(string &str){
    set<char> charSet;
    for (char c : str){
        if (charSet.count(c)){
            return false;
        }
        charSet.insert(c);
    }
    return true;
}

bool isUniqueSorted(string &str){
    sort(str.begin(), str.end());
    for (int i = 1; i < int(str.size()); i++){
        if (str[i] == str[i-1]){
            return false;
        }
    }
    return true;
}

int main() {
    string text = "hello";
    if (isUnique(text)){
        cout << "All characters are unique." << endl;
    } else {
        cout << "There are duplicate characters." << endl;
    }
    if (isUniqueSorted(text)){
        cout << "All characters are unique (sorted check)." << endl;
    } else {
        cout << "There are duplicate characters (sorted check)." << endl;
    }
    return 0;
}